packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -212,3 +216,175 @@ class _TweenAnimationBuilderState<T extends Object?>
     return widget.builder(context, _currentTween!.evaluate(animation), widget.child);
   }
 }
+
+/// Widget that animates an [Animatable] value and repeats indefinitely.
+///
+/// The animation continuously cycles from the [Animatable] evaluated at 0.0 to
+/// the value at 1.0.
+/// When [reverse] is true, the animation reverses direction when it reaches
+/// the end. When [paused] is true, the animation stops at its current value.
+///
+/// {@macro flutter.widgets.TweenAnimationBuilder.optimizations}
+///
+/// {@tool dartpad}
+/// This example shows a continuously rotating square that can be paused and resumed.
+///
+/// ** See code in examples/api/lib/widgets/tween_animation_builder/repeating_tween_animation_builder.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [TweenAnimationBuilder], which animates a tween value once.
+///  * [AnimationController.repeat], the underlying mechanism.
+class RepeatingTweenAnimationBuilder<T extends Object> extends StatefulWidget {
+  /// Creates a widget that repeats an animatable animation.
+  const RepeatingTweenAnimationBuilder({
+    super.key,
+    required this.animatable,
+    required this.duration,
+    this.curve = Curves.linear,
+    this.reverse = false,
+    this.paused = false,
+    required this.builder,
+    this.child,
+  });
+
+  /// The animatable to drive repeatedly.
+  ///
+  /// The animation continuously cycles from the value returned by evaluating
+  /// [animatable] at 0.0 to the value returned at 1.0.
+  /// In most cases this will be a [Tween] or [TweenSequence], but any
+  /// [Animatable] that produces values of type [T] is accepted.
+  ///
+  /// ## Ownership
+  ///
+  /// The [RepeatingTweenAnimationBuilder] takes ownership of the provided
+  /// [Animatable] and may mutate it. Once an [Animatable] instance has been
+  /// passed to [RepeatingTweenAnimationBuilder] its properties should not be
+  /// accessed or changed to avoid interference with the
+  /// [RepeatingTweenAnimationBuilder].
+  ///
+  /// It is good practice to never store an [Animatable] provided to a
+  /// [RepeatingTweenAnimationBuilder] in an instance variable to avoid accidental
+  /// modifications of the [Animatable].
+  final Animatable<T> animatable;
+
+  /// The duration of the animation.
+  final Duration duration;
+
+  /// The curve to apply when animating the value.
+  ///
+  /// Defaults to [Curves.linear].
+  final Curve curve;
+
+  /// Called every time the animation value changes.
+  ///
+  /// The builder receives the current animation value and the optional child.
```

Comment:
I suggest a different first paragraph to introduce this property more directly.
```suggestion
  /// A builder that creates the animated widget subtree.
  ///
  /// Invoked on every frame with the current animation value and the optional
  /// prebuilt child. The child can be incorporated to reuse a subtree that
  /// doesn't depend on the animation, avoiding unnecessary rebuilds.
```

Reply:
Strictly speaking, this might not be called on every frame. For example, things like `Visibility` can mute a running animation.

Maybe:

```suggestion
  /// A builder that creates the animated widget subtree.
  ///
  /// The builder is called every time the animation value changes.
  ///
  /// The optional child can be used to avoid unnecessary rebuilds
  /// if a part of the subtree doesn't depend on the animation.
```